c/driver/postgresql/postgres_copy_reader.h

```diff
@@ -1215,27 +1202,25 @@ class PostgresCopyStreamWriter {
         ArrowArrayViewInitFromSchema(&array_view_.value, schema, nullptr));
     NANOARROW_RETURN_NOT_OK(ArrowArrayViewSetArray(&array_view_.value, array, nullptr));
     root_writer_.Init(&array_view_.value);
+    ArrowBufferInit(&buffer_.value);
     return NANOARROW_OK;
   }
 
-  ArrowErrorCode WriteHeader(ArrowBuffer* buffer, ArrowError* error) {
-    ArrowBufferAppend(buffer, kPgCopyBinarySignature, sizeof(kPgCopyBinarySignature));
+  ArrowErrorCode WriteHeader(ArrowError* error) {
+    ArrowBufferAppend(&buffer_.value, kPgCopyBinarySignature,
```

Comment:
I overlooked this the first time, but shouldn't we be checking the return codes for these calls?